### ISSUE TITLE
unknown token type returned when previous query generates many errors

### ISSUE THREAD

**Issue by janssjo** [denisenkom/go-mssqldb#582]
**Describe the bug**
I get an `unknown token type returned: token(0)` error on a valid query, if I run the valid query immediately after having executed a previous query which generated many errors (seems to happen where there are 10 errors or more). This happens at least when the errors are violations of unique or foreign key constraints.

**To Reproduce**
```
package main

import (
	"database/sql"
	"log"

	_ "github.com/denisenkom/go-mssqldb"
)

func main() {
	connStr := "sqlserver://user:password@server/instance?database=test_db&log=63"

	db, err := sql.Open("sqlserver", connStr)
	if err != nil {
		panic(err)
	}
	defer db.Close()

	_, err = db.Exec(`
		CREATE TABLE #tableA (
			id INT NOT NULL UNIQUE,
		)

		INSERT INTO #tableA (id) VALUES (1);
		INSERT INTO #tableA (id) VALUES (1);
		INSERT INTO #tableA (id) VALUES (1);
		INSERT INTO #tableA (id) VALUES (1);
		INSERT INTO #tableA (id) VALUES (1);
		INSERT INTO #tableA (id) VALUES (1);
		INSERT INTO #tableA (id) VALUES (1);
		INSERT INTO #tableA (id) VALUES (1);
		INSERT INTO #tableA (id) VALUES (1);
		INSERT INTO #tableA (id) VALUES (1);
		INSERT INTO #tableA (id) VALUES (1);
		INSERT INTO #tableA (id) VALUES (1);
	`)
	if err != nil {
		log.Printf("Expected error: %v", err)
	}

	_, err = db.Exec(`
		CREATE TABLE #tableB (
			foo VARCHAR(MAX) NULL
		)
		INSERT INTO #tableB (foo) VALUES ('foo');
	`)
	if err != nil {
		log.Printf("Unexpected error: %v", err)
	}
}
```

The above code produces the following output:
```
2020/06/11 14:30:54
                CREATE TABLE #tableA (
                        id INT NOT NULL UNIQUE,
                )

                INSERT INTO #tableA (id) VALUES (1);
                INSERT INTO #tableA (id) VALUES (1);
                INSERT INTO #tableA (id) VALUES (1);
                INSERT INTO #tableA (id) VALUES (1);
                INSERT INTO #tableA (id) VALUES (1);
                INSERT INTO #tableA (id) VALUES (1);
                INSERT INTO #tableA (id) VALUES (1);
                INSERT INTO #tableA (id) VALUES (1);
                INSERT INTO #tableA (id) VALUES (1);
                INSERT INTO #tableA (id) VALUES (1);
                INSERT INTO #tableA (id) VALUES (1);
                INSERT INTO #tableA (id) VALUES (1);

2020/06/11 14:30:54 (1 row(s) affected)
2020/06/11 14:30:54 Violation of UNIQUE KEY constraint 'UQ__#tableA___3213E83EAC51E13D'. Cannot insert duplicate key in object 'dbo.#tableA'. The duplicate k
ey value is (1).
2020/06/11 14:30:54 The statement has been terminated.
2020/06/11 14:30:54 Violation of UNIQUE KEY constraint 'UQ__#tableA___3213E83EAC51E13D'. Cannot insert duplicate key in object 'dbo.#tableA'. The duplicate k
ey value is (1).
2020/06/11 14:30:54 The statement has been terminated.
2020/06/11 14:30:54 Expected error: mssql: Violation of UNIQUE KEY constraint 'UQ__#tableA___3213E83EAC51E13D'. Cannot insert duplicate key in object 'dbo.#t
ableA'. The duplicate key value is (1).
2020/06/11 14:30:54
                CREATE TABLE #tableB (
                        foo VARCHAR(MAX) NULL
                )
                INSERT INTO #tableB (foo) VALUES ('foo');

2020/06/11 14:30:54 Violation of UNIQUE KEY constraint 'UQ__#tableA___3213E83EAC51E13D'. Cannot insert duplicate key in object 'dbo.#tableA'. The duplicate k
ey value is (1).
2020/06/11 14:30:54 ERROR: Intercepted panic unknown token type returned: token(49)
2020/06/11 14:30:54 ERROR: Intercepted panic unknown token type returned: token(0)
2020/06/11 14:30:54 Unexpected error: unknown token type returned: token(0)
```

**Expected behavior**
The tableA query is expected to produce errors, but the tableB query should succed. I.e. the last three lines in the output, (two `Intercepted panic` and a `unknown token type returned: token(0)` errors) should not happen.

### Further technical details
SQL Server version: 2017 (14.0.3294.2)
SQL Server Operating system: Docker (mcr.microsoft.com/mssql/server:2017-latest)
Host Operating system: Windows 10
github.com/denisenkom/go-mssqldb v0.0.0-20200428022330-06a60b6afbbc

**Additional context**
This happens only if the first query produces 10 or more errors. It doesn't seem to happen for smaller number of errors.

**Comment by HimanshuSharma-Angel** [comment]
any update on this?

**Comment by janssjo** [comment]
This seems to have been fixed in commit 191d776964ad7583dddcf1b9ceaa7be8e5352898. I cannot reproduce this in v0.11.0 or newer.
